### PR TITLE
Xcode 8.0 & Swift 3.0 changes

### PR DESCRIPTION
**Pull request by trooper2013**
Will require everyone to move to Xcode 8 GA. Made a few changes for reducing the warnings.

### CODE REVIEW

**Comment by wmathurin** [comment]
You also need to update the native swift template (https://github.com/forcedotcom/SalesforceMobileSDK-iOS/tree/unstable/build/app_template_files/__NativeSwiftTemplateAppName__)

**Comment by khawkins** [comment]
...aaaand those comments are on a commit, not the PR. 😡  I hate that.  I'll re-add on the PR.

**Comment by khawkins** [comment]
If it says `DevelopmentTeam` or `DEVELOPMENT_TEAM` in the setting of the project file, please remove that line. 😊

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -923,8 +940,10 @@
 			baseConfigurationReference = CE2B8CAB1CE9281F00C6FC6A /* SalesforceAnalytics-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CODE_SIGN_ENTITLEMENTS = SalesforceAnalyticsTestApp/SalesforceAnalyticsTestApp.entitlements;
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
+				DEVELOPMENT_TEAM = "";
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: Remove this entry. You might have to hand edit this file to do so.

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -900,9 +915,11 @@
 			baseConfigurationReference = CE2B8CAA1CE9281F00C6FC6A /* SalesforceAnalytics-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				CODE_SIGN_ENTITLEMENTS = SalesforceAnalyticsTestApp/SalesforceAnalyticsTestApp.entitlements;
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
 				DEBUG_INFORMATION_FORMAT = dwarf;
+				DEVELOPMENT_TEAM = "";
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: Remove this entry. You might have to hand edit this file to do so.

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -759,7 +774,7 @@
 			baseConfigurationReference = CE2B8CAA1CE9281F00C6FC6A /* SalesforceAnalytics-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
 				APPLICATION_EXTENSION_API_ONLY = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: We should revert this to `iPhone Developer`.

File: libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -783,7 +798,7 @@
 			baseConfigurationReference = CE2B8CAB1CE9281F00C6FC6A /* SalesforceAnalytics-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
 				APPLICATION_EXTENSION_API_ONLY = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SalesforceAnalytics/SalesforceAnalytics.xcodeproj/project.pbxproj]: We should revert this to `iPhone Developer`.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1682,7 +1711,7 @@
 				CLANG_WARN_DIRECT_OBJC_ISA_USAGE = YES_ERROR;
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -1721,7 +1750,7 @@
...
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDK.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SalesforceHybridSDK/SalesforceHybridSDKTestApp/SalesforceHybridSDKTestApp-Info.plist [hunk 7]
```diff
@@ -5,7 +5,7 @@
 	<key>CFBundleDevelopmentRegion</key>
 	<string>en</string>
 	<key>CFBundleDisplayName</key>
-	<string>${PRODUCT_NAME}</string>
+	<string>SalesforceHybridSDKTestApp</string>
```
Comment (bhariharan) [inline libs/SalesforceHybridSDK/SalesforceHybridSDKTestApp/SalesforceHybridSDKTestApp-Info.plist]: Isn't `${PRODUCT_NAME}` the same as `SalesforceHybridSDKTestApp`? Why hardcode this here?

File: libs/SalesforceReact/SalesforceReact.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -457,7 +491,7 @@
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SalesforceReact/SalesforceReact.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SalesforceReact/SalesforceReact.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -513,7 +547,7 @@
...
```
Comment (bhariharan) [inline libs/SalesforceReact/SalesforceReact.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -3428,7 +3473,7 @@
 				CLANG_WARN_OBJC_ROOT_CLASS = YES_ERROR;
 				CLANG_WARN_UNREACHABLE_CODE = YES;
 				CLANG_WARN__DUPLICATE_METHOD_MATCH = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -3484,7 +3529,7 @@
...
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SmartStore/SmartStore.xcodeproj/project.pbxproj [hunk 12]
```diff
@@ -1037,7 +1046,7 @@
 			baseConfigurationReference = CECB66231C1BB886008038AE /* SmartStore-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
 				CLANG_WARN_UNREACHABLE_CODE = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SmartStore/SmartStore.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SmartStore/SmartStore.xcodeproj/project.pbxproj [hunk 13]
```diff
@@ -1065,7 +1074,7 @@
 			baseConfigurationReference = CECB66241C1BB886008038AE /* SmartStore-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
 				CLANG_WARN_UNREACHABLE_CODE = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SmartStore/SmartStore.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SmartSync/SmartSync.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -1077,7 +1086,7 @@
 			baseConfigurationReference = CECB662A1C1BB8A9008038AE /* SmartSync-Dynamic-iOS-Debug.xcconfig */;
 			buildSettings = {
 				CLANG_WARN_UNREACHABLE_CODE = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SmartSync/SmartSync.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: libs/SmartSync/SmartSync.xcodeproj/project.pbxproj [hunk 15]
```diff
@@ -1105,7 +1114,7 @@
 			baseConfigurationReference = CECB662B1C1BB8A9008038AE /* SmartSync-Dynamic-iOS-Release.xcconfig */;
 			buildSettings = {
 				CLANG_WARN_UNREACHABLE_CODE = YES;
-				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
+				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "";
```
Comment (bhariharan) [inline libs/SmartSync/SmartSync.xcodeproj/project.pbxproj]: Revert this to `iPhone Developer`.

File: native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj [hunk 16]
```diff
@@ -475,7 +475,7 @@
 				4F1DA68F1C84FD7E0014FAD3 /* Info.plist */,
 			);
 			path = SmartSyncExplorerUITests;
-			sourceTree = "<group>";
+			sourceTree = SOURCE_ROOT;
```
Comment (bhariharan) [inline native/SampleApps/SmartSyncExplorer/SmartSyncExplorer.xcodeproj/project.pbxproj]: Why has this changed? Shouldn't it be `<group>`?

**Pull request merged**